src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -2243,6 +2362,18 @@ public override BoundNode VisitAssignmentOperator(BoundAssignmentOperator node)
             return null;
         }
 
+        private static bool UseLegacyWarnings(BoundExpression expr)
+        {
+            switch (expr.Kind)
```

Comment:
Consider doing an exhaustive enumeration.
I'm assuming that fields are part of API, so use regular (not W) warnings.
For parameters, I think that `ref` and `out` parameters should use regular (not W) warnings as well (because that breaks caller "contract", which is possible safety concern). For by-val and other parameters, maybe W is ok.

Reply:
I agree, assigning to a parameter passed by value could be a W warning while assigning to a `ref` or `out` parameter should be a regular warning. I'll follow up in a subsequent PR.

---
In reply to: [182820706](https://github.com/dotnet/roslyn/pull/26107#discussion_r182820706) [](ancestors = 182820706)